mkosi/backend.py

```diff
@@ -951,3 +957,22 @@ def complete_step(cls, text: str, text2: Optional[str] = None) -> Iterator[List[
 
         if text2 is not None:
             cls.print_step(text2.format(*args))
+
+
+def chown_to_running_user(paths: Union[PathString, [PathString]]):
+    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or '0')
+    user = pwd.getpwuid(uid).pw_name
+    gid = pwd.getpwuid(uid).pw_gid
+
+    # Root already owns generated files/directories, skipping.
+    if uid == 0:
+        return
```

Comment:
I'd go for extra safety and still do the chowning even if it's root, you never know what the actual owner might be